optimizely/client/client_test.go

```diff
@@ -1573,6 +1694,41 @@ func (s *ClientTestSuiteFM) TestIsFeatureEnabled() {
 	s.mockDecisionService.AssertExpectations(s.T())
 }
 
+func (s *ClientTestSuiteFM) TestIsFeatureEnabledWithDecisionError() {
+	testUserContext := entities.UserContext{ID: "test_user_1"}
+
+	// Test happy path
+	testVariation := makeTestVariation("green", true)
+	testExperiment := makeTestExperimentWithVariations("number_1", []entities.Variation{testVariation})
+	testFeature := makeTestFeatureWithExperiment("feature_1", testExperiment)
+	s.mockConfig.On("GetFeatureByKey", testFeature.Key).Return(testFeature, nil)
+	s.mockConfigManager.On("GetConfig").Return(s.mockConfig, nil)
+
+	// Set up the mock decision service and its return value
+	testDecisionContext := decision.FeatureDecisionContext{
+		Feature:       &testFeature,
+		ProjectConfig: s.mockConfig,
+	}
+
+	expectedFeatureDecision := decision.FeatureDecision{
+		Experiment: testExperiment,
+		Variation:  &testVariation,
+		Source:     decision.FeatureTest,
+	}
+
+	s.mockDecisionService.On("GetFeatureDecision", testDecisionContext, testUserContext).Return(expectedFeatureDecision, errors.New(""))
+
+	client := OptimizelyClient{
+		ConfigManager:   s.mockConfigManager,
+		DecisionService: s.mockDecisionService,
+	}
+	result, _ := client.IsFeatureEnabled(testFeature.Key, testUserContext)
```

Comment:
Shoulnd't we assert on the error being returned?

Reply:
addressed